scripts/Test-RepoPackage.ps1

```diff
@@ -0,0 +1,633 @@
+  and before taking the screenshot.
+
+.PARAMETER screenShotDir
+  The directory to where the scrip should save all taken screenshots.
+  (The directory does not need to exist before running the script)
+
+.EXAMPLE
+  .\Test-RepoPackage.ps1
+#>
+param(
+  [string]$packageName = $null,
+  [ValidateSet('all','update','install','uninstall','none')]
+  [string]$type = 'all',
+  [switch]$CleanFiles,
+  [switch]$TakeScreenshots,
+  [int]$chocoCommandTimeout = 600,
+  [int]$timeoutBeforeScreenshot = 1,
+  [string]$screenShotDir = "$env:TEMP\screenshots"
+)
+
+function WriteOutput() {
+<#
+.SYNOPSIS
+  Simple helper function to simplify
+  the different output methods we use.
+#>
+  param(
+    [Parameter(ValueFromPipeline = $true)]
+    [string]$text,
+    [ValidateSet('Error','Warning','ChocoError','ChocoWarning','ChocoInfo','Normal')]
+    [string]$type = 'Normal'
+  )
+   switch -Exact ($type) {
+    'Error' { Write-Error $text }
+    'Warning' { Write-Warning $text }
+    'ChocoError' { Write-Host $text -ForegroundColor Black -BackgroundColor Red }
+    'ChocoWarning' { Write-Host $text -ForegroundColor Black -BackgroundColor Yellow }
+    'ChocoInfo' { Write-Host $text -ForegroundColor White -BackgroundColor Black }
+    Default { Write-Host $text }
+  }
+}
+
+function WriteChocoOutput() {
+<#
+.SYNOPSIS
+  A simple helper function for processing the output of choco.
+  Only meant to handle the colorization of text.
+#>
+  param(
+    [Parameter(ValueFromPipeline = $true)]
+    [string]$text
+  )
+  begin {}
+  process {
+    switch -Regex ($text) {
+      '^\s*ERROR|\s*Failures|^Uninstall may not be silent' { WriteOutput $text -type ChocoError }
+      '^\s*WARNING' { WriteOutput $text -type ChocoWarning }
+      Default { WriteOutput $text -type ChocoInfo }
+    }
+  }
+  end {}
+}
+
+function CleanFiles() {
+<#
+.SYNOPSIS
+  The function responsible for cleaning out temporary
+  files before we run the tests.
+#>
+  param(
+    [Parameter(Mandatory = $true)]
+    [string]$screenShotDir
+  )
+  $pathsToClean = @(
+    "$env:ChocolateyInstall\logs\*"
+    "$env:TEMP\chocolatey\au\*"
+    "$PSScriptRoot\..\Update-Force-Test*.md"
+  )
+  if ($TakeScreenshots) { $pathsToClean += @("$screenShotDir\*") }
+  $pathsToClean += Get-ChildItem -Recurse "$PSScriptRoot\.." -Filter "*.nupkg" | select -Expand FullName
+
+  $pathsToClean | % {
+    if (Test-Path $_) { rm $_ -Recurse }
+  }
+}
+
+function GetDependentPackage() {
+<#
+.SYNOPSIS
+  Function responsible for aquiring a meta package(s) dependency.
+#>
+  param(
+    [Parameter(Mandatory = $true)]
+    [string]$packageDirectory
+  )
+  $packageName = $PackageDirectory -split '[\/\\]' | select -last 1
+  $nuspecPath = Get-Item "$PackageDirectory\*.nuspec"
+  $content = Get-Content -Encoding UTF8 $nuspecPath
+  $Matches = $null
+  $content | ? { $_ -match "\<dependency.*id=`"(${packageName}.(install|portable|app|commandline))" } | select -First 1 | WriteOutput
```

Comment:
true, It could be on multiple lines. I've yet to see I package that have split the dependency element on multiple lines though, so didn't bother using `[xml]` for this as it seemed unnecessary.